drivers/i2c/i2c_shell.c

```diff
@@ -320,10 +328,12 @@ static void device_name_get(size_t idx, struct shell_static_entry *entry)
 SHELL_DYNAMIC_CMD_CREATE(dsub_device_name, device_name_get);
 
 SHELL_STATIC_SUBCMD_SET_CREATE(sub_i2c_cmds,
+#if CONFIG_I2C_SMBUS_SUPPORTED
```

Comment:
Macro name: **_CONFIG_I2C_SMBUS_SUPPORTED_**, is chosen like this because i2c transfer without any data is supported by SMbus.

By default, this macro is enabled and the i2c scan shell command will be available for an execution but if we disable this macro explicitly then the i2c scan shell command will not be available and thus preventing the program from getting hung if  i2c transfers without data are not handled in the driver (ex: i2c_dw driver).